packages/components/src/internal/components/domainproperties/samples/SampleTypePropertiesPanel.tsx

```diff
@@ -336,6 +356,39 @@ class SampleTypePropertiesPanelImpl extends React.PureComponent<
                                 />
                             </Col>
                         </Row>
+                        {includeMetricUnitProperty &&
+                            <Row className="margin-top">
+                                <Col xs={2}>
+                                    <DomainFieldLabel
+                                        label={metricUnitLabel}
+                                        required={metricUnitRequired}
+                                        helpTipBody={() => metricUnitHelpMsg}
+                                    />
+                                </Col>
+                                <Col xs={3}>
+                                    {metricUnitOptions ?
+                                        <SelectInput
+                                            formsy={false}
+                                            containerClass="sampleset-metric-unit-select-container"
+                                            inputClass="sampleset-metric-unit-select"
+                                            name="metricUnit"
+                                            options={metricUnitOptions}
+                                            onChange={(name, formValue, option) => {this.onFieldChange(name, formValue === undefined && option ? option.id : formValue)}}
+                                            placeholder={`Select a unit...`}
+                                            value={model.metricUnit}
+                                        /> :
+                                        <FormControl
```

Comment:
Since the fixed set of units are freezermanager specific,  I think there might be need to support free-text for other products.

Reply:
OK.  It's sort of a tradeoff at this point since we don't have a use for this field that doesn't involve the FM application, but if the option is already wired up we can leave it.